File: src/com/reco1l/framework/net/IDownloaderObserver.java
package com.reco1l.framework.net;

/**
 * @author Reco1l
 */
public interface IDownloaderObserver
{

    default void onDownloadStart(Downloader downloader)
    {
    }

    default void onDownloadEnd(Downloader downloader)
    {
    }

    default void onDownloadCancel(Downloader downloader)
    {
    }

    default void onDownloadUpdate(Downloader downloader)
    {
    }

    default void onDownloadFail(Downloader downloader)
    {
    }
}

File: src/com/reco1l/framework/net/IDownloaderObserver.kt
package com.reco1l.framework.net

/**
 * @author Reco1l
 */
interface IDownloaderObserver
{
    fun onDownloadStart(downloader: Downloader)
    {
    }

    fun onDownloadEnd(downloader: Downloader)
    {
    }

    fun onDownloadCancel(downloader: Downloader)
    {
    }

    fun onDownloadUpdate(downloader: Downloader)
    {
    }

    fun onDownloadFail(downloader: Downloader)
    {
    }
}
